spec/providers/firestore.spec.ts

```diff
@@ -145,7 +145,10 @@ describe('Firestore Functions', () => {
       let testFunction = firestore.document('path').onCreate((event) => {
         expect(event.data.data()).to.deep.equal({key1: true, key2: 123});
         expect(event.data.get('key1')).to.equal(true);
-        expect(event.data.previous).to.equal(null);
+        expect(event.data.previous).to.not.equal(null);
+        expect(event.data.previous.exists).to.be.false;
+        expect(event.data.previous.data).to.throw(Error);
```

Comment:
Why isn't event.data.previous.data just null?

Reply:
cc/ @schmidt-sebastian 

Lauren's implementation matches the client implementation (thanks!) but we are actually intending to change this so that .data is null for nonexistent documents (see http://b/68270401).  So I think it would probably be fine (and future looking even) to make .data null here. I believe the difficulty that the clients have in making this change (that DocumentSnapshots are used to surface query results where the document is guaranteed to exist and therefore .data should not be nullable) doesn't apply to Functions.

Reply:
@mikelehen `event.data.previous` is a constructed DocumentSnapshot. So I didn't do any additional processing to it. Once the Firestore Node SDK makes .data null, this will be null as well.

Reply:
Ah, right! I forgot we're sharing code (yay!).

Reply:
Ooof. Should we make the unit test support both cases since it will otherwise randomly break in CI once the latest node module is picked up (assuming it's considered a revisional fix and not a breaking change)